src/adapters/node.js

```diff
@@ -1,36 +1,53 @@
-// Generated by CoffeeScript 1.9.3
-(function() {
+(function () {
     var request = require('request');
     var mkFhir = require('../fhir');
     var Q = require('q');
 
     var adapter = {
         defer: Q.defer,
-        http: function(args) {
+        http: function (args) {
             var deff = Q.defer();
-            args.body = args.data;
+
+            if (args.resource != undefined) {
+                args.body = args.resource
+            }
             // url should be relative to baseUrl.
-            if(args.url) {
+            if (args.url) {
                 args.url = args.url.replace(args.baseUrl, '');
+            } 
+            // does not like that for a transaction
+            if (args.url == "") {
+                args.url = "/";
             }
             args.json = true;
-            if(args.debug){
-                console.log('DEBUG[node]: (requrest)', args); 
+            if (args.debug) {
+                console.log('DEBUG[node]: (request)', args);
             }
-            request(args, function(err, response, body) {
-                var headers = function(x) {return response.headers[x.toLowerCase()];};
-                var resp = {data: body, status: response.statusCode, headers: headers, config: args};
-                if(args.debug){
-                    console.log('DEBUG[node]: (responce)', resp); 
+            request(args, function (err, response, body) {
+                // making headers congruent to native.js
+                var headers = undefined;
+                var statusCode = undefined;
+                if (response != undefined) {
+                    headers = {
+                        resp: response.headers,
```

Comment:
Can you clarify why a new property is beings created here, as opposed to the previous approach of referencing the request variable?

Reply:
In native.js the response headers are added via the following code

>      // This object is in the shape required by fhir.js lib
>        Object.assign(returnableObject, {
>          status: response.status,
>          headers: response.headers,
>          config: args,
>        });

response.headers is typed as:  readonly headers: Headers; (lib.es6.d.ts: mmmh is this my setup or this is correct for native.js?) and Headers has a get function defined: get(name: string): string | null;

this is why I added the get() function to the returned object. but maybe I'm wrong and I was assuming the wrong header type. however I could not locate any test or example where response headers are extracted and how the interface should be used. do you have any opinion on this?